src/services/mapService.js

```diff
@@ -0,0 +1,63 @@
+export const mapService = {
+  mapSettings: {
+    defaultWidth: 360,
+    defaultHeight: 180,
+    scale: 1.5
+  },
+
+  getMapElement(){
+    return document.getElementById("map");
+  },
+
+  getMapWidth() {
+    return this.mapSettings.defaultWidth * this.mapSettings.scale;
+  },
+
+  getMapHeight(){
+    return this.mapSettings.defaultHeight * this.mapSettings.scale;
+  },
+
+  translateLongitudeToCanvasPx(longitude){
+    let x = (longitude + this.mapSettings.defaultWidth / 2) * this.mapSettings.scale;
+    return x;
+  },
+
+  translateLatitudeToCanvasPx(latitude){
+    let y = ((this.mapSettings.defaultHeight / 2) - latitude) * this.mapSettings.scale;
+    return y;
+  },
+
+  initMap(){
+    const mapCanvas = this.getMapElement();
+    const ctx = mapCanvas.getContext("2d");
+    ctx.canvas.width = this.getMapWidth();
+    ctx.canvas.height = this.getMapHeight();
+
+    // draw some coffee shops
+    this.drawCoffeeShopAt(ctx, this.translateLongitudeToCanvasPx(25), this.translateLatitudeToCanvasPx(45));
+    this.drawCoffeeShopAt(ctx, this.translateLongitudeToCanvasPx(150), this.translateLatitudeToCanvasPx(30));
+    this.drawCoffeeShopAt(ctx, this.translateLongitudeToCanvasPx(-100), this.translateLatitudeToCanvasPx(50));
+    this.drawCoffeeShopAt(ctx, this.translateLongitudeToCanvasPx(-60), this.translateLatitudeToCanvasPx(-30));
```

Comment:
Now, I am foreseeing some issues in the future for this. When you will be working with the API, it will return an array of coordinates. When that happens, you will have to rewrite this function that is only responsible for displaying the map. This is a sign that the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) is broken.
The data layer (what coffee shops you get, and from where) should be separated from the display layer (this function), rewrite this function to accept an array of coffee shops. When you will integrate with the API, you will just change the data source and this function will stay the same.